### ISSUE TITLE
Attached shell doesn' t support some keys

### ISSUE THREAD

**Issue by xgdgsc** [Microsoft/vscode-docker#480]
When I attach a shell to docker container, I cannot use some common shell keys like `tab` for completion, `up` and `down` to switch to history commands.

**Comment by StephenWeatherford** [comment]
What OS are you using, and also what terminal shell (e.g. PowerShell, bash...).  I'm having no trouble on Mac in the bash shell.

Although honestly we don't have any control over this kind of behavior.  All we're doing is running the command in the terminal.

**Comment by xgdgsc** [comment]
On Ubuntu 18.04, should be bash.

**Comment by xgdgsc** [comment]
I found the difference, the attached shell runs command `docker exec -it HASH /bin/sh`
well working one runs `docker exec -it HASH /bin/bash` . Can I change the attached shell running command?

**Comment by xgdgsc** [comment]
https://github.com/Microsoft/vscode-docker/blob/master/commands/open-shell-container.ts#L20 changing this default to `/bin/bash`?

**Comment by xgdgsc** [comment]
https://unix.stackexchange.com/a/182871/38063

**Comment by StephenWeatherford** [comment]
Thanks for figuring that out. Just change the docker.attachShellCommand.linuxContainer configuration setting in vscode and you should be good to go!

**Comment by xgdgsc** [comment]
OK. Thanks. Would you consider making `/bin/bash` the default?

**Comment by StephenWeatherford** [comment]
I've actually wondered about that myself.

@chrisdias Was there a reason we're using /bin/sh instead of /bin/bash?  Is /bin/sh more available?  Thx.  @fiveisprime

**Comment by chrisdias** [comment]
nothing that i can remember, except a familiarity with `/bin/bash`.

**Comment by StephenWeatherford** [comment]
Will move to default of bash.

**Comment by StephenWeatherford** [comment]
Need to change the default back to sh.  We've been seeing images where bash does not exist.  In particular, alpine doesn't have bash.  See #118 for an example.

**Comment by StephenWeatherford** [comment]
@xgdgsc @fiveisprime How do you feel about using the following as the default value?
```
    /bin/sh -c "[ -e /bin/bash ] && /bin/bash || /bin/sh"
```
e.g.
![image](https://user-images.githubusercontent.com/6913354/48103112-ae6c6d00-e1e2-11e8-9467-227f5dbee8c3.png)

Not sure if there's a simpler solution - will execute bash if it exists, otherwise sh.

**Comment by xgdgsc** [comment]
That looks good.

**Comment by PrashanthCorp** [comment]
@fiveisprime  approves.

**Comment by PrashanthCorp** [comment]
As a note for Windows git bash users, I can confirm that Stephen's suggestion works on powershell, cmd, and bash(Mac). It does _not_ work on git bash(Windows). For this, please escape the leading '/' : 
```
//bin/sh -c "[ -e /bin/bash ] && /bin/bash || /bin/sh"
```

Hope this helps.

**Comment by StephenWeatherford** [comment]
Fixed (again) via #622
